adapters/nativery/nativery.go

```diff
@@ -0,0 +1,302 @@
+	for _, sb := range nativeryResponse.SeatBid {
+		for i := range sb.Bid {
+			bid := sb.Bid[i]
+
+			// should be data sended from nativery server to partecipate to the auction
+			var bidExt bidExt
+			if err := json.Unmarshal(bid.Ext, &bidExt); err != nil {
+				errs = append(errs, err)
+				continue
+			}
+
+			bidType, err := getMediaTypeForBid(&bidExt)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			// get metadata
+			bidMeta := buildBidMeta(string(bidType), bidExt.Nativery.BidAdvDomains)
+
+			bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: bidType,
+				// metadata is encouraged
+				BidMeta: bidMeta,
+			})
+		}
+
+	}
+
+	// set bidder currency, EUR by default
+	if nativeryResponse.Cur != "" {
+		bidderResponse.Currency = nativeryResponse.Cur
+	} else {
+		bidderResponse.Currency = "EUR"
+	}
+	return bidderResponse, errs
+
+}
+
+// getMediaTypeForBid switch nativery type in bid type.
+func getMediaTypeForBid(bid *bidExt) (openrtb_ext.BidType, error) {
+	switch bid.Nativery.BidType {
+	case "native":
+		return openrtb_ext.BidTypeNative, nil
+	case "display", "banner", "rich_media":
+		return openrtb_ext.BidTypeBanner, nil
+	case "video":
+		return openrtb_ext.BidTypeVideo, nil
+	default:
+		return "", fmt.Errorf("unrecognized bid_ad_media_type in response from nativery: %s", bid.Nativery.BidType)
+	}
+}
+
+func convertIntToBoolean(num *int) bool {
+	var b bool
+	// Dereferenzia num usando *
+	if num != nil && *num == 1 {
+		b = true
+	} else {
+		b = false
+	}
+	return b
+}
+
+func buildBidMeta(mediaType string, advDomain []string) *openrtb_ext.ExtBidPrebidMeta {
+
+	//advertiserDomains and dchain are encouraged to implements
+	return &openrtb_ext.ExtBidPrebidMeta{
+		MediaType:         mediaType,
+		AdvertiserDomains: advDomain,
+		/*
+			DChain: json.RawMessage{} ,
+			Cosa include Dchain:
+				nodes: Un array di oggetti che rappresentano i diversi partecipanti alla catena di domanda.
+				complete: Un flag che indica se la catena di domanda è completa (1) o incompleta (0).
+				ver: La versione del modulo Dchain utilizzato.
+		*/
+	}
+}
+
+func splitRequests(imps []openrtb2.Imp, request *openrtb2.BidRequest, requestExt map[string]json.RawMessage, requestExtNativery bidReqExtNativery, uri string) ([]*adapters.RequestData, []error) {
+	var errs []error
+
+	resArr := make([]*adapters.RequestData, 0, 1)
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	nativeryExtJson, err := json.Marshal(requestExtNativery)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	requestExtClone := maps.Clone(requestExt)
+	requestExtClone["nativery"] = nativeryExtJson
+
+	request.Ext, err = json.Marshal(requestExtClone)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	for _, imp := range imps {
+		impsForReq := []openrtb2.Imp{imp}
+		request.Imp = impsForReq
```

Comment:
@andreafassina : this line is overriding the previously set imps objects with single imp object per run. So, it's actually modifying the imp of  `reqCopy` object  which is passed from outside. So, let ensure that after `splitRequests` call `reqCopy.imp` is not referred anywhere, which is expecting all the imps